babynames.py

```diff
@@ -14,48 +14,32 @@
 import re
 import argparse
 
-"""
-Define the extract_names() function below and change main()
-to call it.
-
-For writing regex, it's nice to include a copy of the target
-text for inspiration.
-
-Here's what the html looks like in the baby.html files:
-...
-<h3 align="center">Popularity in 1990</h3>
-....
-<tr align="right"><td>1</td><td>Michael</td><td>Jessica</td>
-<tr align="right"><td>2</td><td>Christopher</td><td>Ashley</td>
-<tr align="right"><td>3</td><td>Matthew</td><td>Brittany</td>
-...
-
-Suggested milestones for incremental development:
- -Extract the year and print it
- -Extract the names and rank numbers and just print them
- -Get the names data into a dict and print it
- -Build the [year, 'name rank', ... ] list and print it
- -Fix main() to use the extract_names list
-"""
-
-
 def extract_names(filename):
     """
     Given a file name for baby.html, returns a list starting with the year string
     followed by the name-rank strings in alphabetical order.
     ['2006', 'Aaliyah 91', Aaron 57', 'Abagail 895', ' ...]
     """
-    # +++your code here+++
-    return
+    with open(filename, 'r') as file:
+        baby_names = []
+        read_file = file.read()
+        match_object = re.search(r'Popularity in (\d{4})', read_file)
+        match_baby_object = re.findall(r'<td>(\d*)</td><td>(\w+)</td><td>(\w+)</td>', read_file)
+        if match_object:
+            baby_year = match_object.group(1)
+            baby_names.append(baby_year)
+        if match_baby_object:
+            for tup in match_baby_object:
```

Comment:
You can use automatic tuple unpacking to assign tuple members to a variable to avoid index access: https://realpython.com/python-lists-tuples/#tuple-assignment-packing-and-unpacking